indico/modules/rb_new/client/js/modules/calendar/actions.js

```diff
@@ -16,30 +16,53 @@
  */
 
 import fetchCalendarURL from 'indico-url:rooms_new.calendar';
-import {indicoAxios} from 'indico/utils/axios';
+import searchRoomsURL from 'indico-url:rooms_new.search_rooms';
+
+import _ from 'lodash';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
 import {ajaxAction} from 'indico/utils/redux';
 import {preProcessParameters} from '../../util';
-import {ajax as ajaxRules} from './serializers';
+import {ajaxRules as roomSearchAjaxRules} from '../../common/roomSearch';
 
 
-export const SET_DATE = 'calendar/SET_DATE';
 export const ROWS_RECEIVED = 'calendar/ROWS_RECEIVED';
 export const FETCH_REQUEST = 'calendar/FETCH_REQUEST';
 export const FETCH_SUCCESS = 'calendar/FETCH_SUCCESS';
 export const FETCH_ERROR = 'calendar/FETCH_ERROR';
+export const ROOM_IDS_RECEIVED = 'calendar/ROOM_IDS_RECEIVED';
 
 
-export function setDate(date) {
-    return {type: SET_DATE, date};
-}
-
-export function fetchCalendar() {
+export function fetchCalendar(onlyDateChanged = false) {
     return async (dispatch, getState) => {
-        const {calendar: {data}} = getState();
-        const params = preProcessParameters(data, ajaxRules);
+        dispatch({type: FETCH_REQUEST});
+        const {calendar: {filters}, calendar: {data: {roomIds}}} = getState();
+        let newRoomIds = roomIds;
+
+        if (!onlyDateChanged) {
+            newRoomIds = null;
+            if (!_.isEmpty(filters)) {
```

Comment:
What about splitting this into a separate function (not-exported, simply in here above or below `fetchCalendar`).

Then you can simplify this block to something like

    newRoomIds = _.isEmpty(filters) ? null : await fetchRoomIds(filters)